### DISCUSSION TITLE
I cannot use \includegraphics{}

### DISCUSSION THREAD

**Discussion by shiyq2018** [MiKTeX/miktex#1061] [Q&A]
I use \includegraphics{fig1} in TextWorks, but am stuck with the error that fig_1-eps-converted-to.pdf is not generated. What's the reason for that? \usepackage{graphicx} is included. I tried to add --shell-escape, but it still doesn't work. How can I proceed? Thanks a lot!

**Comment by ivankokan** [comment]
> I use \includegraphics{fig1} in TextWorks, but am stuck with the error that fig_1-eps-converted-to.pdf is not generated. What's the reason for that? \usepackage{graphicx} is included. I tried to add --shell-escape, but it still doesn't work. How can I proceed? Thanks a lot!

TextWorks -> TeXworks
Provide the MWE and the logs.

**Reply by shiyq2018** [reply]
**Thanks for your correction.

The relevant codes are as follows:**


\documentclass[%
reprint,
amsmath,amssymb,
]{revtex4-2}

\usepackage{graphicx}% Include figure files

\begin{figure}[b]
\includegraphics{fig_1}% Here is how to import EPS art
\caption{\label{fig:epsart} A figure caption. The figure captions are
automatically numbered.}
\end{figure}

**The fig_1.eps is in the same folder as tex file. 

The log is as follows,**

Package epstopdf Info: Source file: <fig_1.eps>
(epstopdf)                    date: 2020-10-03 17:26:56
(epstopdf)                    size: 193 bytes
(epstopdf)             Output file: <fig_1-eps-converted-to.pdf>
(epstopdf)             Command: <miktex-epstopdf --outfile=fig_1-eps-converted-
to.pdf fig_1.eps>
(epstopdf)             \includegraphics on input line 485.
runsystem(miktex-epstopdf --outfile=fig_1-eps-converted-to.pdf fig_1.eps)...exe
cuted safely (allowed).

Package epstopdf Info: Result file: <fig_1-eps-converted-to.pdf>.

! Package pdftex.def Error: File `fig_1-eps-converted-to.pdf' not found: using 
draft setting.

See the pdftex.def package documentation for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.485 \includegraphics{fig_1}
                             % Here is how to import EPS art
? H
Try typing  <return>  to proceed.
If that doesn't work, type  X <return>  to quit.
? 
Package epstopdf Info: Source file: <fig_2.eps>
(epstopdf)                    date: 2020-10-03 17:26:56
(epstopdf)                    size: 196 bytes
(epstopdf)             Output file: <fig_2-eps-converted-to.pdf>
(epstopdf)             Command: <miktex-epstopdf --outfile=fig_2-eps-converted-
to.pdf fig_2.eps>
(epstopdf)             \includegraphics on input line 501.
runsystem(miktex-epstopdf --outfile=fig_2-eps-converted-to.pdf fig_2.eps)...exe
cuted safely (allowed).

Package epstopdf Info: Result file: <fig_2-eps-converted-to.pdf>.

! Package pdftex.def Error: File `fig_2-eps-converted-to.pdf' not found: using 
draft setting.

See the pdftex.def package documentation for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.501 \includegraphics{fig_2}
                             % Here is how to import EPS art
? X
 
Here is how much of TeX's memory you used:

**Comment by ivankokan** [comment]
I added the `\listfiles` at the end of the preamble and successfully compiled the code:
```
pdflatex --shell-escape blabla
```

Here is my log with up-to-date MiKTeX:

```
This is pdfTeX, Version 3.141592653-2.6-1.40.24 (MiKTeX 22.1) (preloaded format=pdflatex.fmt)
 \write18 enabled.
entering extended mode
(blabla.tex
LaTeX2e <2021-11-15> patch level 1
L3 programming layer <2022-02-21>
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/revtex\revtex4-2.cls
Document Class: revtex4-2 2020/10/03 4.2e (https://journals.aps.org/revtex/ for
 documentation)
ltxutil[2020/10/03 4.2e utilities package (portions licensed from W. E. Baxter 
web at superscript.com)]
ltxfront[2020/10/03 4.2e frontmatter package (AO,DPC,MD)]
ltxgrid[2020/10/03 4.2e page grid package (portions licensed from W. E. Baxter 
web at superscript.com)]

Class revtex4-2 Warning: No Society specified, using default society aps.

(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/revtex\aps4-2.rtx)

Class revtex4-2 Warning: No journal specified, using default pra.


(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/revtex\aps10pt4-2.rtx
)
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/textcase\textcase.sty
)
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/amsfonts\amssymb.sty
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/amsfonts\amsfonts.sty
))
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/amsmath\amsmath.sty
For additional information on amsmath, use the `?' option.
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/amsmath\amstext.sty
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/amsmath\amsgen.sty))
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/amsmath\amsbsy.sty)
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/amsmath\amsopn.sty))
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/url\url.sty)
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/natbib\natbib.sty)
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/revtex\revsymb4-2.sty
))
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/graphics\graphicx.sty
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/graphics\keyval.sty)
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/graphics\graphics.sty
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/graphics\trig.sty)
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/graphics-cfg\graphics
.cfg)
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/graphics-def\pdftex.d
ef)))
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/l3backend\l3backend-p
dftex.def) (blabla.aux)
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/context/base/mkii\supp-pdf.
mkii
[Loading MPS to PDF converter (version 2006.09.02).]
)
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/epstopdf-pkg\epstopdf
-base.sty
(C:\Users\testuser\AppData\Local\Programs\MiKTeX\tex/latex/00miktex\epstopdf-sys
.cfg))

pdfTeX warning: pdflatex (file ./fig_1-eps-converted-to.pdf): PDF inclusion: fo
und PDF version <1.7>, but at most version <1.5> allowed

Overfull \hbox (175.57396pt too wide) in paragraph at lines 15--16
 [][] 
No file blabla.bbl.
[1{C:/Users/testuser/AppData/Local/MiKTeX/fonts/map/pdftex/pdftex.map} <./fig_1-
eps-converted-to.pdf>] (blabla.aux)

 *File List*
revtex4-2.cls    2020/10/03 4.2e (https://journals.aps.org/revtex/ for document
ation)
  aps4-2.rtx
aps10pt4-2.rtx
textcase.sty    2019/09/14 v1.00 Text only upper/lower case changing (DPC)
 amssymb.sty    2013/01/14 v3.01 AMS font symbols
amsfonts.sty    2013/01/14 v3.01 Basic AMSFonts support
 amsmath.sty    2021/10/15 v2.17l AMS math features
 amstext.sty    2021/08/26 v2.01 AMS text
  amsgen.sty    1999/11/30 v2.0 generic functions
  amsbsy.sty    1999/11/29 v1.2d Bold Symbols
  amsopn.sty    2021/08/26 v2.02 operator names
     url.sty    2013/09/16  ver 3.4  Verb mode for urls, etc.
  natbib.sty    2010/09/13 8.31b (PWD, AO)
revsymb4-2.sty    2020/10/03 4.2e (https://journals.aps.org/revtex/ for documen
tation)
graphicx.sty    2021/09/16 v1.2d Enhanced LaTeX Graphics (DPC,SPQR)
  keyval.sty    2014/10/28 v1.15 key=value parser (DPC)
graphics.sty    2021/03/04 v1.4d Standard LaTeX Graphics (DPC,SPQR)
    trig.sty    2021/08/11 v1.11 sin cos tan (DPC)
graphics.cfg    2016/06/04 v1.11 sample graphics configuration
  pdftex.def    2020/10/05 v1.2a Graphics/color driver for pdftex
l3backend-pdftex.def    2022-02-07 L3 backend support: PDF output (pdfTeX)
supp-pdf.mkii
epstopdf-base.sty    2020-01-24 v2.11 Base part for package epstopdf
epstopdf-sys.cfg    2021/03/18 v2.0 Configuration of epstopdf for MiKTeX
fig_1-eps-converted-to.pdf    Graphic file (type pdf)
 ***********

 )
(see the transcript file for additional information)<C:/Users/testuser/AppData/L
ocal/Programs/MiKTeX/fonts/type1/public/amsfonts/cm/cmr10.pfb><C:/Users/testuser
/AppData/Local/Programs/MiKTeX/fonts/type1/public/amsfonts/cm/cmr9.pfb>
Output written on blabla.pdf (1 page, 31838 bytes).
Transcript written on blabla.log.
```

**Reply by shiyq2018** [reply]
Hi, thanks for your demonstration. But the problem still exists for me. I re-checked the log file and it's:

`Sorry, but "MiKTeX EPS-to-PDF Converter" did not succeed.

The log file hopefully contains the information to get MiKTeX going again:

  /Users/xxx/Library/Application Support/MiKTeX/texmfs/data/miktex/log/miktex-epstopdf.log`

and I go to the MiKTeX-epstopdf.log field, there's error as follows:

`2022-02-25 18:54:32,878+0100 INFO  miktex-epstopdf - this process (2594) started by miktex-pdftex in directory /Users/xxx/Documents/aps with command line: miktex-epstopdf --outfile=fig_1-eps-converted-to.pdf fig_1.eps
2022-02-25 18:54:32,980+0100 FATAL miktex-epstopdf.core - Ghostscript could not be not found.
2022-02-25 18:54:32,980+0100 FATAL miktex-epstopdf.core - Data: 
2022-02-25 18:54:32,980+0100 FATAL miktex-epstopdf.core - Source: Libraries/MiKTeX/Core/Session/gsinfo.cpp:97
2022-02-25 18:54:32,983+0100 FATAL miktex-epstopdf - Ghostscript could not be not found.
2022-02-25 18:54:32,984+0100 FATAL miktex-epstopdf - Info: 
2022-02-25 18:54:32,984+0100 FATAL miktex-epstopdf - Source: Libraries/MiKTeX/Core/Session/gsinfo.cpp
2022-02-25 18:54:32,984+0100 FATAL miktex-epstopdf - Line: 97
`

Why it says Cannot find ghoscript, eventhough I checked the MikTex control that the ghoscript package is actually installed.

BTW, how do you type the command line to compile in Texworks? I can only click the green button to compile pdfLaTex. And I add the "--shell-escape" in the Preferences -> Typesetting -> pdfLatex edit. I assume it's the same, isn't it?

I checked all the files and I think I have all the files as you listed.
